metadatastore/commands.py

```diff
@@ -345,34 +557,69 @@ def insert_event(descriptor, time, seq_num, data, timestamps=None, uid=None):
     (not stored, but exchanged) timestamps is optional. If not included,
     data is expected to include (value, timestamp) tuples.
     """
-    if timestamps is not None and set(data.keys()) != set(timestamps.keys()):
-        raise ValueError("The fields in 'data' and 'timestamps' must match.")
-    if timestamps is not None:
-        # new Document exchange format
-        val_ts_tuple = _transform_data(data, timestamps)
-    else:
-        # old Document exchange format
-        val_ts_tuple = data
+    val_ts_tuple = _transform_data(data, timestamps)
 
     # Allow caller to beg forgiveness rather than ask permission w.r.t
     # EventDescriptor creation.
     if descriptor is None:
         raise EventDescriptorIsNoneError()
```

Comment:
Do we need this exception since descriptor a positional? Also the code that used this is dead in ophyd (and since it was my implementation, I can state with confidence that it was a bad design anyway)